src/main/java/wooteco/subway/ui/LineController.java

```diff
@@ -0,0 +1,65 @@
+package wooteco.subway.ui;
+
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import wooteco.subway.dao.LineDao;
+import wooteco.subway.domain.Line;
+import wooteco.subway.dto.LineRequest;
+import wooteco.subway.dto.LineResponse;
+
+import java.net.URI;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+@RestController
+public class LineController {
+    private final LineDao lineDao;
+
+    public LineController(LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
+
+    @PostMapping("/lines")
+    public ResponseEntity<LineResponse> createLine(@RequestBody LineRequest lineRequest) {
+        Line Line = new Line(lineRequest.getName(), lineRequest.getColor());
+        Line newLine = lineDao.save(Line);
+
+        LineResponse lineResponse = new LineResponse(newLine.getId(), newLine.getName(), newLine.getColor());
+        return ResponseEntity.created(URI.create("/lines/" + newLine.getId())).body(lineResponse);
+    }
+
+    @GetMapping(value = "/lines", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<LineResponse>> showLines() {
+        List<Line> lines = lineDao.findAll();
+        List<LineResponse> lineResponses = lines.stream()
+                .map(it -> new LineResponse(it.getId(), it.getName(), it.getColor()))
+                .collect(Collectors.toList());
+        return ResponseEntity.ok().body(lineResponses);
+    }
+
+    @GetMapping("/lines/{id}")
+    public ResponseEntity<LineResponse> searchLine(@PathVariable Long id) {
+
+        Line line = lineDao.findById(id);
+        LineResponse lineResponse = new LineResponse(line.getId(), line.getName(), line.getColor());
+        return ResponseEntity.ok().body(lineResponse);
```

Comment:
```suggestion
    public LineResponse searchLine(@PathVariable Long id) {

        Line line = lineDao.findById(id);
        LineResponse lineResponse = new LineResponse(line.getId(), line.getName(), line.getColor());
        return lineResponse;
```
상태 코드가 항상 200 OK 라면 조금 더 간단하게 표현할 수 있습니다.

Reply:
오 무조건 ResponseEntity에 담아서 반환해야 되는건 아니군요. 200 일 때만 가능한 건가요?? 찾아보겠습니다.